iac-extensions/kubernetes/src/main/java/org/sonar/iac/kubernetes/plugin/HelmProcessor.java

```diff
@@ -51,38 +53,50 @@ public void initialize() {
     }
   }
 
-  @CheckForNull
+  public boolean isHelmEvaluatorInitialized() {
+    return helmEvaluator != null;
+  }
+
   String processHelmTemplate(String filename, String source, InputFileContext inputFileContext) {
-    if (helmEvaluator != null) {
-      // TODO: better support of Helm project structure
-      var sourceWithComments = HelmPreprocessor.addLineComments(source);
-      var valuesFile = findValuesFile(inputFileContext);
-      if (valuesFile != null) {
-        try {
-          return evaluateHelmTemplate(filename, sourceWithComments, valuesFile.contents());
-        } catch (IOException e) {
-          LOG.debug("Failed to read values file at {}, skipping processing of Helm file '{}'", valuesFile, inputFileContext.inputFile, e);
-        }
-      } else {
-        LOG.debug("Failed to find values file, skipping processing of Helm file '{}'", inputFileContext.inputFile);
-      }
+    if (helmEvaluator == null) {
+      throw new IllegalStateException("Attempt to process Helm template with uninitialized Helm evaluator");
     }
 
-    return null;
+    // TODO: better support of Helm project structure
+    var sourceWithComments = HelmPreprocessor.addLineComments(source);
+    var valuesFile = findValuesFile(inputFileContext);
+    var valuesFileContent = validateAndReadValuesFile(valuesFile, inputFileContext.inputFile);
+    return evaluateHelmTemplate(filename, inputFileContext.inputFile, sourceWithComments, valuesFileContent);
   }
 
-  @CheckForNull
-  private String evaluateHelmTemplate(String path, String content, String valuesFileContent) {
+  private static String validateAndReadValuesFile(@Nullable InputFile valuesFile, InputFile inputFile) {
+    if (valuesFile == null) {
+      throw new ParseException("Failed to find values file, skipping processing of Helm file " + inputFile,
+        inputFile.newPointer(0, 0), null);
+    }
+
+    String valuesFileContent;
+    try {
+      valuesFileContent = valuesFile.contents();
+    } catch (IOException e) {
+      throw new ParseException("Failed to read values file at " + valuesFile + " while evaluating Helm file " + inputFile,
```

Comment:
I would change this message to be in line with the other exception ("skipping processing of Helm file...").
This way you could create a method
```private void throwParseExceptionFor(InputFile inputFile, String customMessagePrefix, @Nullable String cause)```
as you are always raising on the same range and nearly with the same message.
WDYT?

Reply:
Good suggestion!